cfn-resources/network-peering/cmd/resource/resource.go

```diff
@@ -249,11 +226,15 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	if errEvent := validateModel(DeleteRequiredFields, currentModel); errEvent != nil {
 		return *errEvent, nil
 	}
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %s", err)
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+
+	if currentModel.Profile == nil {
```

Comment:
Stopping here, can we update all the usages using the example above with both checks.